engine/src/main/java/org/pentaho/di/www/GetRegisteredStepsServlet.java

```diff
@@ -0,0 +1,133 @@
+import java.util.List;
+
+/**
+ *
+ *This class contains api's to fetch registered steps.
+ *
+ * @since 2024
+ * @author sVarma1992
+ *
+ */
+public class GetRegisteredStepsServlet extends BaseHttpServlet implements CartePluginInterface {
+
+  private static Class<?> PKG = GetRegisteredStepsServlet.class; // for i18n purposes, needed by Translator2!!
+
+  private static final long serialVersionUID = 3634806745372015720L;
+
+  public static final String CONTEXT_PATH = "/kettle/registeredSteps";
+
+  private static final String PLUGIN_TYPE = "pluginType";
+
+  @VisibleForTesting
+  CarteStatusCache cache = CarteStatusCache.getInstance();
+
+  public GetRegisteredStepsServlet() {
+  }
+
+  public GetRegisteredStepsServlet( TransformationMap transformationMap ) {
+    super( transformationMap );
+  }
+
+  public void doGet( HttpServletRequest request, HttpServletResponse response ) throws ServletException,
+          IOException {
+
+    if ( isJettyMode() && !request.getContextPath().startsWith( CONTEXT_PATH ) ) {
+      return;
+    }
+    response.setStatus( HttpServletResponse.SC_OK );
+
+    response.setCharacterEncoding( "UTF-8" );
+    response.setContentType( "application/json;charset=UTF-8" );
+    String pluginType = request.getParameter( PLUGIN_TYPE );
+
+    PrintWriter out = response.getWriter();
+    PluginRegistry registry = PluginRegistry.getInstance();
+    if ( request.getParameter( "stepId" ) == null ) {
+      writeJson( pluginType, out, registry );
+    } else {
+      String stepId = request.getParameter( "stepId" );
+      PluginInterface pluginInterface = registry.getPlugin( StepPluginType.class, stepId );
+      try {
+        StepMetaInterface meta = (StepMetaInterface) registry.loadClass( pluginInterface );
+        out.print( new ObjectMapper().writeValueAsString( meta ) );
+      } catch ( KettlePluginException e ) {
+        e.printStackTrace();
+      }
+
+    }
+  }
+
+  private void writeJson( String pluginType, PrintWriter out, PluginRegistry registry ) throws JsonProcessingException {
+
+    JSONArray previewJson = new JSONArray();
+    List<PluginInterface> stepPlugins = pluginType == null ? registry.getPlugins( StepPluginType.class )
+            : registry.getPlugins( registry.getPluginTypes().stream()
+            .filter( x -> x.getName().contains( pluginType ) )
+            .findAny().get() );
+    for ( PluginInterface pluginInterface : stepPlugins ) {
+      if ( pluginInterface.getIds().length > 0 ) {
+        JSONObject stepJson = new JSONObject();
+        stepJson.put( "id", pluginInterface.getIds()[0] );
+        stepJson.put( "name", pluginInterface.getName() );
+        stepJson.put( "category", pluginInterface.getCategory() );
+        previewJson.add( stepJson );
+      }
+    }
+    JSONObject finalJsonOutput = new JSONObject();
+    finalJsonOutput.put( "plugins", previewJson );
+    out.print( finalJsonOutput );
+  }
+
+  public String toString() {
+    return "Registered Plugins Handler";
```

Comment:
same here like in other class. if this is common across multiple classes pls make it a constant at util level.